java/src/test/java/com/xxAMIDOxx/xxSTACKSxx/menu/api/v1/impl/DeleteMenuControllerImplTest.java

```diff
@@ -0,0 +1,73 @@
+package com.xxAMIDOxx.xxSTACKSxx.menu.api.v1.impl;
+
+import com.microsoft.azure.spring.autoconfigure.cosmosdb.CosmosAutoConfiguration;
+import com.microsoft.azure.spring.autoconfigure.cosmosdb.CosmosDbRepositoriesAutoConfiguration;
+import com.xxAMIDOxx.xxSTACKSxx.menu.domain.Menu;
+import com.xxAMIDOxx.xxSTACKSxx.menu.repository.MenuRepository;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.xxAMIDOxx.xxSTACKSxx.menu.domain.MenuHelper.createMenu;
+import static com.xxAMIDOxx.xxSTACKSxx.util.TestHelper.getBaseURL;
+import static java.util.UUID.randomUUID;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.http.HttpStatus.OK;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@EnableAutoConfiguration(
+        exclude = {
+                CosmosDbRepositoriesAutoConfiguration.class,
+                CosmosAutoConfiguration.class
+        })
+@Tag("Integration")
+class DeleteMenuControllerImplTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @MockBean
+    private MenuRepository repository;
+
+    @Test
+    void testDeleteSuccess() {
+        // Given
+        Menu menu = createMenu(1);
+        when(repository.findById(eq(menu.getId()))).thenReturn(Optional.of(menu));
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        var requestEntity = new HttpEntity<>(headers);
+        var response =
+                this.testRestTemplate.exchange(String.format("%s/v1/menu/%s", getBaseURL(port), menu.getId()),
+                        HttpMethod.DELETE,
+                        requestEntity, ResponseEntity.class);
+        // Then
+        ArgumentCaptor<Menu> captor = ArgumentCaptor.forClass(Menu.class);
+        verify(repository, times(1)).deleteById(menu.getId());
+        then(response.getStatusCode()).isEqualTo(OK);
+    }
+
+
+}
```

Comment:
We should start thinking about idempotency.
i.e. If you do the same request multiple times you get the same result.
We should probably just return success if the menu to be deleted does not exist. That's something for discussion though.

Reply:
Agree, the tests should be idempotent. It'd be good to have a `tearDown` method to remove the menu, etc if the test scenario fails.